src/AccessibilityInsights.SharedUx/Dialogs/ContainedDialog.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using AccessibilityInsights.CommonUxComponents.Controls;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Automation.Peers;
+using System.Windows.Controls;
+using System.Windows.Threading;
+
+namespace AccessibilityInsights.SharedUx.Dialogs
+{
+    public abstract class ContainedDialog : UserControl
+    {
+        /// <summary>
+        /// Used to determine when the "dialog" is ready to return its result
+        /// </summary>
+        protected EventWaitHandle WaitHandle { get; } = new AutoResetEvent(true);
+
+        protected bool DialogResult { get; set; } = false;
```

Comment:
There is a known dialog result type to indicate many different return values. how about using that type here than "bool"? 
also having it in the base class would be good.

Reply:
You might be thinking of the [Win Forms DialogResult](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.dialogresult?view=netframework-4.7.2). I'm following the [Window.ShowDialog pattern](https://docs.microsoft.com/en-us/dotnet/api/system.windows.window.showdialog?view=netframework-4.7.2), which returns `bool?`. This is what wwe use for our own `MessageDialog` class as well. I could make it a nullable bool for the sake of consistency, but even that doesn't seem necessary to me.

Reply:
I see. it makes sense.